src/history/hs_rec.c

```diff
@@ -519,36 +497,35 @@ __wt_hs_insert_updates(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_MULTI *mult
              */
             if (upd->type == WT_UPDATE_STANDARD && F_ISSET(upd, WT_UPDATE_HS))
                 break;
+
+            /*
+             * Save the first non timestamped update in the update chain. This is used to reset all
+             * the following update timestamps in the chain. Do not save if it is the last update in
+             * the chain. While inserting this last update, all the existing history store updates
+             * will reinsert after correcting their timestamps.
+             */
+            if (non_ts_upd == NULL && upd->start_ts == WT_TS_NONE && upd->next != NULL) {
```

Comment:
It's possible the next is an aborted and it is the last update in the chain. Does that impact anything?

Reply:
I think so, it can cause problems. Fixed it by verifying all the possible flows to reset the pointer.

Reply:
I don't understand the fix for this. If we have U@20->U@10->T@0->A, the non_ts_upd is set to T@0, but the first update insert to hs is U@10. When do we reset the pointer in this case?

Reply:
The first update to insert is `A`. Whenever the `A` time window is formed, at that time the `prev_upd` is pointed to `T`, then by comparing it we will reset.

In the scenario, if the U@0 is present, whenever it is inserted into the history store, the pointer gets reset.

or if T@0 is the last one in the list, we remove all the existing history store entries, after that the pointer get reset.

In total 3 places, the pointer gets reset.
1. T@0 is the last place.
2. U@0 is getting inserted into the history store.
3. T@0 in the middle, while inserting the previous update, it gets reset.

Reply:
No A is aborted so we never insert it.